src/compiler/commandLineParser.ts

```diff
@@ -1089,7 +1089,7 @@ const commandOptionsWithoutBuild: CommandLineOption[] = [
     {
         name: "allowImportingTsExtensions",
         type: "boolean",
-        affectsModuleResolution: true,
+        affectsSemanticDiagnostics: true,

```

Comment:
This was a mistake—the option has very intentionally never affected how module resolution works, only whether an error is raised. I’m not sure whether it caused the issue we observed during the bug bash, but I added a unit test that shows the error disappearing when updating this option.